From david-b@pacbell.net  Tue Feb 24 21:09:07 2009
From: Anand Gadiyar <gadiyar@ti.com>
Date: Sat, 21 Feb 2009 15:31:40 -0800
Subject: USB: musb: resume suspended root hub on disconnect
To: Greg KH <greg@kroah.com>
Cc: linux-usb@vger.kernel.org, Anand Gadiyar <gadiyar@ti.com>
Message-ID: <200902211531.40766.david-b@pacbell.net>
Content-Disposition: inline


From: Anand Gadiyar <gadiyar@ti.com>

If this is not done, khubd will not be informed of the disconnect
and will assume the device is still there.

Easily seen when a hub is connected with no device attached to it;
it will autosuspend.  When the hub is disconnected, it still shows
up in /proc/bus/usb/devices

Signed-off-by: Anand Gadiyar <gadiyar@ti.com>
Acked-by: Felipe Balbi <felipe.balbi@nokia.com>
Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/musb_core.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/musb/musb_core.c
+++ b/drivers/usb/musb/musb_core.c
@@ -767,6 +767,7 @@ static irqreturn_t musb_stage2_irq(struc
 #ifdef CONFIG_USB_MUSB_HDRC_HCD
 		case OTG_STATE_A_HOST:
 		case OTG_STATE_A_SUSPEND:
+			usb_hcd_resume_root_hub(musb_to_hcd(musb));
 			musb_root_disconnect(musb);
 			if (musb->a_wait_bcon != 0)
 				musb_platform_try_idle(musb, jiffies
